x-pack/metricbeat/module/openai/usage/config.go

```diff
@@ -0,0 +1,88 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package usage
+
+import (
+	"errors"
+	"fmt"
+	"time"
+)
+
+type Config struct {
+	APIKeys    []apiKeyConfig   `config:"api_keys" validate:"required"`
+	APIURL     string           `config:"api_url" validate:"required"`
+	Headers    []string         `config:"headers"`
+	RateLimit  *rateLimitConfig `config:"rate_limit"`
+	Timeout    time.Duration    `config:"timeout" validate:"required"`
+	Collection collectionConfig `config:"collection"`
+}
+
+type rateLimitConfig struct {
+	Limit *int `config:"limit" validate:"required"`
+	Burst *int `config:"burst" validate:"required"`
+}
+
+type apiKeyConfig struct {
+	Key string `config:"key" validate:"required"`
+}
+
+type collectionConfig struct {
+	LookbackDays int  `config:"lookback_days" validate:"required"`
+	Realtime     bool `config:"realtime"`
+}
+
+func defaultConfig() Config {
+	return Config{
+		APIURL:  "https://api.openai.com/v1/usage",
+		Timeout: 30 * time.Second,
+		RateLimit: &rateLimitConfig{
+			Limit: ptr(60),
+			Burst: ptr(5),
+		},
+		Collection: collectionConfig{
+			LookbackDays: 0,     // 0 days
+			Realtime:     false, // avoid realtime collection by default
+		},
+	}
+}
+
+func (c *Config) Validate() error {
+	var errs []error
+
+	if len(c.APIKeys) == 0 {
+		errs = append(errs, errors.New("at least one API key must be configured"))
+	}
+	if c.APIURL == "" {
+		errs = append(errs, errors.New("api_url cannot be empty"))
```

Comment:
Should we also add a check for API URL for correct format along with the nil check ?

if _, err := url.ParseRequestURI(c.APIURL); err != nil {
    errs = append(errs, fmt.Errorf("invalid api_url: %v", err))
}

Reply:
Yes, we can. I'll add.